### DISCUSSION TITLE
Adding Dogecoin package to Gentoo Guru

### DISCUSSION THREAD

**Discussion by victorsk2019** [dogecoin/dogecoin#3321] [General]
Hi everyone,

Thought I'd revisit some work I did with Gentoo ebuilds couple of years ago to add Dogecoin to Gentoo package lists.  It has been suggested that it be added to Guru overlay instead.  I have recently updated the ebuilds I worked on and only have recent current live version (no in-development versions like I had earlier).  The advantage of installing Dogecoin using package manager for Gentoo system users is that all required dependencies and code fixes are applied automatically, so, for example, issues encountered in this report: https://github.com/dogecoin/dogecoin/issues/3210 (I also encountered similar issue during installation of Qt component), are automatically addressed using patches during installation process for the user.

Please, feel free to examine the ebuild and patches in `net-p2p/dogecoin-qt` package I made in my repository: https://github.com/victorsk2019/gentoo-guru/tree/master/net-p2p/dogecoin-qt
I tested installing `dogecoin-qt-1.14.6` on authentic Gentoo system and on Gentoo-based Calculate Linux and it installs beautifully, applying patches and setting desktop menu option in Plasma DE 👌

I will greatly appreciate any feedback and can make a PR into Gentoo Guru if there are no problems.  I'll also work on adding `dogecoind` for full node implementation. 

Thank you!
Victor.

**Comment by victorsk2019** [comment]
Just a quick note: I've submitted new dogecoin-qt package into Gentoo Guru's `dev` branch [here](https://github.com/gentoo/guru/tree/dev/net-p2p/dogecoin-qt).  Please, feel free to comment on any of the ebuild's aspects or reach out directly on the email address in the metadata.xml, if there are any concerns 😊. 

Yay! 🎉

![image](https://github.com/dogecoin/dogecoin/assets/47955466/a5cdec17-c415-459b-842a-e5fc41b91e3b)

**Comment by victorsk2019** [comment]
Hi,

Small question: is `scrypt-sse2` option still going to be 'experimental' with 1.14.7 release?  I've added this option to 1.14.7 locally (please see below) as part of `CPU_FLAGS_X86` flags, however, if this feature is planned to be experimental with 1.14.7 production release, then maybe I'll call it something like `sse2-experimental` in the USE flag instead.  Any thoughts on adding 1.14.7 to Guru overlay in an unkeyworded/testing phase as shown below?
  
 <img width="1368" alt="Screenshot 2023-07-30 at 11 10 49 AM" src="https://github.com/dogecoin/dogecoin/assets/47955466/7f3b5bbf-d306-4e8c-a1e8-ed04ba4a3b64">

P.S. thank you @patricklodder for clarification earlier.

**Comment by patricklodder** [comment]
> is scrypt-sse2 option still going to be 'experimental' with 1.14.7 release? 

As it stands right now, yes. If tomorrow there is a security issue and we have to do a release, we'll have to keep it in experimental. Personally I would like both `scrypt-sse2` and `armv8-crypto` to mature out of experimental with an upcoming release but it still needs work to get there.

**Comment by victorsk2019** [comment]
> I would like both scrypt-sse2 and armv8-crypto to mature out of experimental with an upcoming release but it still needs work to get there.

Well, I ran into a small issue using/adding `armv8-crypto` for Gentoo Guru project.  It seems config is modifying CXXFLAGS by adding `march` value  `armv8-a+crypto` when selecting `armv8-crypto` so this could be a problem with Gentoo policy against modifying CXXFLAGS.  However, it turns out my Raspberry Pi 400 BCM2835 doesn't support cryptographic instructions (I kept getting "illegal instruction" error when building with `crypto` selection).  It is too bad because based on what I have read `armv8-crypto` could drastically boost cryptographic algorithm computation, and subsequently, Dogecoin's performance.  Does Apple's M1 processor have crypto extensions?

**Comment by patricklodder** [comment]
Re: RPi. That doesn't have crypto support. A fix is in the making in #3188 in the form of runtime activation.

Re: CXXFLAGS. Please have a look at build-aux/m4, configure.ac and Makefile.am - we do tons of CXXFLAGS modification (local to the build of course, not global.) So if this is an issue then you may need to not use the autoconf from this repo at all. Maybe look at how LukeJr does it for Bitcoin?

**Comment by victorsk2019** [comment]
Thank you! https://github.com/dogecoin/dogecoin/pull/3188 subject is exactly what I've been working with lately.  I was looking for a PR on this subject but couldn't find.

My RPI CPU arm flags:
`cpuid2cpuflags`
`CPU_FLAGS_ARM: edsp neon thumb vfp vfpv3 vfpv4 vfp-d32 crc32 v4 v5 v6 v7 v8 thumb2`
Tried to use `neon` for the build but it didn't work.

BTW please feel free to close this issue.  Though I think it may be useful information for those considering getting RPI and the fact that it doesn't have crypto support.  Had I seen discussion in PR #3188 earlier, for example, I probably would have bought another ARM-based model that supports crypto.  Although I'm pretty happy with my RPI 400 now that I have it 😎
